test/Microsoft.AspNet.Mvc.TagHelpers.Test/EnvironmentTagHelperTest.cs

```diff
@@ -0,0 +1,124 @@
+        [InlineData("Staging,Development,Production")]
+        [InlineData("Staging,Production,Development")]
+        public void ShowsContentWhenCurrentEnvironmentIsSpecified(string namesAttribute)
+        {
+            ShouldShowContent(namesAttribute);
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData(null)]
+        [InlineData("  ")]
+        [InlineData(", ")]
+        [InlineData("   , ")]
+        [InlineData(",")]
+        [InlineData(",,")]
+        [InlineData(",,,")]
+        [InlineData(",,, ")]
+        public void ShowsContentWhenNoEnvironmentIsSpecified(string namesAttribute)
+        {
+            ShouldShowContent(namesAttribute);
+        }
+
+        [Theory]
+        [InlineData("NotDevelopment")]
+        [InlineData("NOTDEVELOPMENT")]
+        [InlineData("NotDevelopment,AlsoNotDevelopment")]
+        [InlineData("Doesn'tMatchAtAll")]
+        [InlineData("Development and a space")]
+        [InlineData("Development and a space,SomethingElse")]
+        public void DoesNotShowContentWhenCurrentEnvironmentIsNotSpecified(string namesAttribute)
+        {
+            var content = "content";
+            var context = MakeTagHelperContext(
+                attributes: new Dictionary<string, object> { { "names", namesAttribute } },
+                content: content);
+            var output = MakeTagHelperOutput("environment");
+            var hostingEnvironment = new Mock<IHostingEnvironment>();
+            hostingEnvironment.SetupProperty(h => h.EnvironmentName);
+            hostingEnvironment.Object.EnvironmentName = "Development";
+
+            var helper = new EnvironmentTagHelper
+            {
+                HostingEnvironment = hostingEnvironment.Object,
+                Names = namesAttribute
+            };
+            helper.Process(context, output);
+
+            Assert.Null(output.TagName);
+            Assert.Null(output.PreContent);
+            Assert.Null(output.Content);
+            Assert.Null(output.PostContent);
+            Assert.True(output.ContentSet);
+        }
+
+        private void ShouldShowContent(string namesAttribute)
+        {
+            var content = "content";
+            var context = MakeTagHelperContext(
+                attributes: new Dictionary<string, object> { { "names", namesAttribute } },
+                content: content);
+            var output = MakeTagHelperOutput("environment");
+            var hostingEnvironment = new Mock<IHostingEnvironment>();
+            hostingEnvironment.SetupProperty(h => h.EnvironmentName);
+            hostingEnvironment.Object.EnvironmentName = "Development";
+
+            var helper = new EnvironmentTagHelper
+            {
+                HostingEnvironment = hostingEnvironment.Object,
+                Names = namesAttribute
+            };
+            helper.Process(context, output);
+
+            Assert.Null(output.TagName);
+            Assert.False(output.ContentSet);
```

Comment:
won't `PreContent`, `Content`, and `PostContent` also be `null` in these scenarios?  if yes, suggest a single test method with `expectedContentSet` values in the data set.